kit/transport/nats/consumer.go

```diff
@@ -0,0 +1,223 @@
+		conn *natn.Conn
+		opts *natn.Options
+		cfg  *natn.ConsumerConfig
+		info *natn.ConsumerInfo
+		js   natn.JetStreamContext
+
+		errorHandler ConsumerErrorHandler
+	}
+)
+
+func WithDurable(durable string) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.Durable = durable
+	}
+}
+
+func WithDeliverSubject(deliverSubject string) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.DeliverSubject = deliverSubject
+	}
+}
+
+func WithDeliverGroup(deliverGroup string) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.DeliverGroup = deliverGroup
+	}
+}
+
+func WithDeliverPolicy(deliverPolicy natn.DeliverPolicy) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.DeliverPolicy = deliverPolicy
+	}
+}
+
+func WithOptStartSeq(optStartSeq uint64) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.OptStartSeq = optStartSeq
+	}
+}
+
+func WithOptStartTime(optStartTime *time.Time) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.OptStartTime = optStartTime
+	}
+}
+
+func WithAckPolicy(ackPolicy natn.AckPolicy) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.AckPolicy = ackPolicy
+	}
+}
+
+func WithAckWait(ackWait time.Duration) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.AckWait = ackWait
+	}
+}
+
+func WithMaxDeliver(maxDeliver int) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.MaxDeliver = maxDeliver
+	}
+}
+
+func WithFilterSubject(filterSubject string) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.FilterSubject = filterSubject
+	}
+}
+
+func WithReplayPolicy(replayPolicy natn.ReplayPolicy) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.ReplayPolicy = replayPolicy
+	}
+}
+
+func WithRateLimit(rateLimit uint64) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.RateLimit = rateLimit
+	}
+}
+
+func WithSampleFrequency(sampleFrequency string) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.SampleFrequency = sampleFrequency
+	}
+}
+
+func WithMaxAckPending(maxAckPending int) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.MaxAckPending = maxAckPending
+	}
+}
+
+func WithFlowControl(flowControl bool) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.FlowControl = flowControl
+	}
+}
+
+func WithHeartbeat(heartbeat time.Duration) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.Heartbeat = heartbeat
+	}
+}
+
+func WithMaxRequestBatch(maxRequestBatch int) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.MaxRequestBatch = maxRequestBatch
+	}
+}
+
+func WithMaxRequestExpires(maxRequestExpires time.Duration) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.MaxRequestExpires = maxRequestExpires
+	}
+}
+
+func WithInactiveThreshold(inactiveThreshold time.Duration) ConsumerOption {
+	return func(c *Consumer) {
+		c.cfg.InactiveThreshold = inactiveThreshold
+	}
+}
+
+func defaultConsumerErrorHandler(cx context.Context, err error) error {
+	return err
+}
+
+func NewConsumer(connstr string, stream string, options ...ConsumerOption) (*Consumer, error) {
```

Comment:
Can you keep this in line with rest of the NATS implementation?
Either change NATS to call it Consumer everywhere, i.e. existing implementation as well or change this to `NewStreamSubscriber()`